src/App.jsx

```diff
@@ -14,45 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_NAME:
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+
+      case SORT_FIELD_LENGTH:
+        preparedGoods
+          .sort((good1, good2) => good1[sortField] - good2[sortField]);
+        break;
+
+      default:
+        return 0;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
```

Comment:
```suggestion
    preparedGoods.reverse();
```
As in every function call we create a copy inside this function you don't need to the result of reverse to `preparedGoods`, you can just reverse